tests/array_split.py

```diff
@@ -22,127 +22,110 @@
-    # test the split routines on 3D array [1:10, 1:10, 1:10] w/ integers,
-    # list of indicies. vsplit, hsplit, dsplit are included
+    # test the split routines on empty, singleton, 2D and 3D arrays
+    # w/ integers, list of indicies. vsplit, hsplit, dsplit are included
     # in the following loops(axis = 0: vsplit, 1: hsplit, 2: dsplit)
-    for i in range(min_size, max_size):
-        for j in range(min_size, max_size):
-            for k in range(min_size, max_size):
-                a = np.random.randn(i, j, k)
-                for axis in range(a.ndim):
-                    input_arr = []
-                    even_div = None  # a.shape[axis]
-                    uneven_div = None  # a.shape[axis]
-                    for div in range(1, (int)(math.sqrt(a.shape[axis]) + 1)):
-                        if a.shape[axis] % div == 0:
-                            even_div = div
-                        else:
-                            uneven_div = div
-                        if even_div is not None and uneven_div is not None:
-                            break
-                    if even_div is None:
-                        even_div = a.shape[axis]
-                    # divisible integer
-                    input_arr.append(even_div)
-                    # indivisble integer
-                    if even_div != uneven_div and uneven_div is not None:
-                        input_arr.append(uneven_div)
-                    # integer larger than shape[axis]
-                    input_arr.append(a.shape[axis] + np.random.randint(1, 10))
-                    # indices array which has points
-                    # within the target dimension of the src array
-                    if a.shape[axis] > 1:
-                        input_arr.append(
-                            list(range(1, a.shape[axis], even_div))
-                        )
-                    # indices array which has points
-                    # out of the target dimension of the src array
-                    input_arr.append(
-                        list(
-                            range(
-                                0,
-                                a.shape[axis]
-                                + even_div * np.random.randint(1, 10),
-                                even_div,
-                            )
-                        )
+
+    input_shapes = [(0), (1), (1, 1), (1, 1, 1), (dim, dim), (dim, dim, dim)]
+    for shape in input_shapes:
+        a = np.random.randint(low=0, high=100, size=shape)
+        for axis in range(a.ndim):
+            input_arr = []
+            even_div = None  # a.shape[axis]
+            uneven_div = None  # a.shape[axis]
+            for div in range(1, (int)(math.sqrt(a.shape[axis]) + 1)):
+                if a.shape[axis] % div == 0:
+                    even_div = div
+                else:
+                    uneven_div = div
+                if even_div is not None and uneven_div is not None:
```

Comment:
I don't think I'm a big fan of this condition, as the values that would be chosen for `even_div` and `uneven_div` are somewhat unpredictable. I feel we're trying to be too smart here, as we won't really run this test with non-default test parameters.

Reply:
I see. This tests take any size of matrix so the dimension could be prime or not, which I tried to handle that. 
What do you mean by 'non-defualt' test parameters?

Reply:
I mean, practically no one will try to run this unit test with the size overriden by the `--dim_size` flag. I'm saying the test is overly complicated for cases that would happen very rarely.